# This file is part of HiGFXback

# requires
REQUIRES="egl-registry-build make-build"

PKG_CONFIG_PATH=/drm/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
sh ./Configure --prefix=/drm

# build
make -C xml

# install
install -d $DESTDIR/drm/include/GL
install -m 644 api/GL/gl.h $DESTDIR/drm/include/GL
install -m 644 api/GL/glext.h $DESTDIR/drm/include/GL
install -d $DESTDIR/drm/include/GLES
install -m 644 api/GLES/gl.h $DESTDIR/drm/include/GLES
install -m 644 api/GLES/glext.h $DESTDIR/drm/include/GLES
install -m 644 api/GLES/glplatform.h $DESTDIR/drm/include/GLES
install -d $DESTDIR/drm/include/GLES2
install -m 644 api/GLES2/gl2.h $DESTDIR/drm/include/GLES2
install -m 644 api/GLES2/gl2ext.h $DESTDIR/drm/include/GLES2
install -m 644 api/GLES2/gl2platform.h $DESTDIR/drm/include/GLES2
install -d $DESTDIR/drm/include/GLES3
install -m 644 api/GLES3/gl3.h $DESTDIR/drm/include/GLES3
install -m 644 api/GLES3/gl3platform.h $DESTDIR/drm/include/GLES3
install -d $DESTDIR/drm/lib/pkgconfig
install -m 644 opengl-registry.pc $DESTDIR/drm/lib/pkgconfig
install -d $DESTDIR/drm/share/opengl/registry
install -m 644 xml/gl.xml $DESTDIR/drm/share/opengl/registry

# build.pc
install -d $DESTDIR/drm/share/pkgconfig
cat > $DESTDIR/drm/share/pkgconfig/opengl-registry-build.pc << EOF
Name: OpenGL-Registry
Version: 20210922
Description: OpenGL API Registry
Requires: $REQUIRES

devel=\\
/drm/include/GL/gl.h \\
/drm/include/GL/glext.h \\
/drm/include/GLES/gl.h \\
/drm/include/GLES/glext.h \\
/drm/include/GLES/glplatform.h \\
/drm/include/GLES2/gl2.h \\
/drm/include/GLES2/gl2ext.h \\
/drm/include/GLES2/gl2platform.h \\
/drm/include/GLES3/gl3.h \\
/drm/include/GLES3/gl3platform.h \\
/drm/lib/pkgconfig/opengl-registry.pc \\
/drm/share/opengl/registry/gl.xml
EOF

exit

# patch
--- OpenGL-Registry-20210922.orig/Configure
+++ OpenGL-Registry-20210922/Configure
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+prefix=/usr/local
+
+while [ $# != 0 ]; do
+  case $1 in
+    --prefix=*) prefix=${1#*=};;
+    *) echo "Unknown option: '$1'"; exit 1;;
+  esac
+  shift
+done
+
+sed "/^prefix/s|=.*|=$prefix|" xml/opengl-registry.pc.in > opengl-registry.pc
--- OpenGL-Registry-20210922.orig/xml/Makefile
+++ OpenGL-Registry-20210922/xml/Makefile
@@ -6,7 +6,7 @@
 
 PYFILES = genheaders.py reg.py
 GENOPTS =
-GENHEADERS = ./genheaders.py $(GENOPTS)
+GENHEADERS = python3 ./genheaders.py $(GENOPTS)
 
 # Generate all headers for GL / GLES / WGL / GLX
 # Different headers depend on different XML registry files
@@ -19,8 +19,9 @@
 # platform-specific versions of these headers, where appropriate.
 
 API	   = ../api
-GLHEADERS  = $(API)/GL/glext.h \
-	     $(API)/GL/glcorearb.h \
+GLHEADERS  = \
+	     $(API)/GL/gl.h \
+	     $(API)/GL/glext.h \
 	     $(API)/GLES/gl.h \
 	     $(API)/GLES/glext.h \
 	     $(API)/GLES2/gl2.h \
--- OpenGL-Registry-20210922.orig/xml/genheaders.py
+++ OpenGL-Registry-20210922/xml/genheaders.py
@@ -56,6 +56,7 @@
 
 allVersions       = allExtensions = '.*'
 noVersions        = noExtensions = None
+glthrough45Pat    = '1|2|3|4.[0-5]'
 gl12andLaterPat   = '1\.[2-9]|[234]\.[0-9]'
 gles2onlyPat      = '2\.[0-9]'
 gles2through30Pat = '2\.[0-9]|3\.0'
@@ -191,6 +192,25 @@
 protectProto = protect
 
 buildList = [
+    # GL compatibility profile - GL/gl.h
+    CGeneratorOptions(
+        filename          = '../api/GL/gl.h',
+        apiname           = 'gl',
+        profile           = 'compatibility',
+        versions          = allVersions,
+        emitversions      = glthrough45Pat,
+        defaultExtensions = None,
+        addExtensions     = None,
+        removeExtensions  = None,
+        prefixText        = prefixStrings + glExtPlatformStrings + genDateCommentString,
+        genFuncPointers   = True,
+        protectFile       = protectFile,
+        protectFeature    = protectFeature,
+        protectProto      = False,
+        protectProtoStr   = 'GL_GLEXT_PROTOTYPES',
+        apicall           = 'GLAPI ',
+        apientry          = 'APIENTRY ',
+        apientryp         = 'APIENTRYP '),
     # GL API 1.2+ + extensions - GL/glext.h
     CGeneratorOptions(
         filename          = '../api/GL/glext.h',
--- OpenGL-Registry-20210922.orig/xml/opengl-registry.pc.in
+++ OpenGL-Registry-20210922/xml/opengl-registry.pc.in
@@ -0,0 +1,8 @@
+prefix=@prefix@
+datarootdir=${prefix}/share
+headersdir=${pc_sysrootdir}${prefix}/include
+pkgdatadir=${pc_sysrootdir}${datarootdir}/opengl/registry
+
+Name: OpenGL Registry
+Description: OpenGL API registry file
+Version: 20210922

# source
https://github.com/KhronosGroup/OpenGL-Registry/archive/b9e1d1a1b37bb064a6d0e3fceffa8aeab5713a38/OpenGL-Registry-20210922.tar.gz
